### PR TITLE
1. Use model13.text.tar.gz ; 2. fix table.conf with database format entry s...

### PR DESCRIPTION
**Pull request by marguerite**
…hipped with libpinyin 2.4.92

### CODE REVIEW

**Comment by wengxt** [comment]
No. you should not use the libpinyin's data file. Even if you want to replace the data, you should use the one from https://github.com/libzhuyin/ibus-libzhuyin/blob/9bb4588fd2c59c48e10e799112ef5e7f307a31b2/data/model/Makefile.am

File: data/CMakeLists.txt [hunk 1]
```diff
@@ -18,8 +18,8 @@ endforeach()
 
 configure_file(${CMAKE_CURRENT_SOURCE_DIR}/table.conf ${CMAKE_CURRENT_BINARY_DIR}/table.conf COPYONLY)
 
-set(ZHUYIN_DATA_FILE_NAME model.text.20161206.tar.gz)
-set(ZHUYIN_URL "http://download.fcitx-im.org/data/${ZHUYIN_DATA_FILE_NAME}")
+set(ZHUYIN_DATA_FILE_NAME model19.text.tar.gz)
+set(ZHUYIN_URL "http://downloads.sourceforge.net/libpinyin/models/${ZHUYIN_DATA_FILE_NAME}")
```
Comment (wengxt) [inline data/CMakeLists.txt]: check sum is not updated, so this doesn't work at all.

File: CMakeLists.txt [hunk 2]
```diff
@@ -23,6 +23,7 @@ endif ()
 pkg_check_modules(LibZhuyin REQUIRED IMPORTED_TARGET "libzhuyin>=2.3.0")
 
 add_definitions(-DFCITX_GETTEXT_DOMAIN=\"fcitx5-zhuyin\")
+add_definitions(-DDATABASE_FORMAT=\"KyotoCabinet\")
```
Comment (wengxt) [inline CMakeLists.txt]: This is not the right way to work with configure_file
set(DATABASE_FORMAT "KyotoCabinet") will do.

But this is a configure option for libpinyin. This is how libzhuyin detect the option:
https://github.com/libzhuyin/ibus-libzhuyin/blob/9bb4588fd2c59c48e10e799112ef5e7f307a31b2/configure.ac#L67

**Review by wengxt — CHANGES_REQUESTED** [review]

**Comment by marguerite** [comment]
@csslayer Thanks!

1. I thought libzhuyin has been merged into libpinyin so they can rely on the same data. I’ll fix 

2. libzhuyin itself was merged into libpinyin, and libpinyin 2.4.92 enforces a database format check when calling gen_binary_files, which can either be “BerkeleyDB” or “KyotoCabinet”.  So I think it’s not a libpinyin configure option but a new mandatory table.conf entry:

https://github.com/libpinyin/libpinyin/commit/5e6c16f7fe72408fd0a96087f8513ca70732dd93

**Comment by wengxt** [comment]
I made the requested change for you, FYI.

Also, I won't use libzhuyin's data. The default for nihao is not even 你好 but 尼號, that's somewhat unacceptable to me.

**Pull request merged**